lld/wasm/InputChunks.cpp

```diff
@@ -378,7 +378,7 @@ void InputChunk::generateRelocationCode(raw_ostream &os) const {
     uint64_t offset = getVA(rel.Offset) - getInputSectionOffset();
 
     Symbol *sym = file->getSymbol(rel);
-    if (!ctx.isPic && sym->isDefined())
+    if (!ctx.isPic && !sym->hasGOTIndex())
```

Comment:
Is this change needed as part of this change?   I think this line could probably use a comment.   i.e. Why are we skipping these relocations?

Reply:
Should this change be differed to #108146 since it seems like a separate semantic change perhaps?

Reply:
this change is necessary because this function is now called regardless of unresolved policy.

Reply:
I'm finding this clauses a little hard to grok without a lot of introspection.

Maybe we could add a comment here?  Something like:

`// Runtime relocations are needed when we don't know the address of a symbol statically.`

Or better still, we could add a variable to make it more clear what is going on:

```
bool requiresRuntimeReloc = ctx.isPic || sym->hasGOTIndex();
if (!requiresRuntimeReloc)
   continue;
```

Reply:
> I'm finding this clauses a little hard to grok without a lot of introspection.

i agree!

> Maybe we could add a comment here? Something like:
> 
> `// Runtime relocations are needed when we don't know the address of a symbol statically.`
> 
> Or better still, we could add a variable to make it more clear what is going on:
> 
> ```
> bool requiresRuntimeReloc = ctx.isPic || sym->hasGOTIndex();
> if (!requiresRuntimeReloc)
>    continue;
> ```

applied. thank you.